File: VB/VB/WinWebSolution.Module.Web/WebShowCustomFormWindowController.vb
Imports Microsoft.VisualBasic
Imports System.Web.UI
Imports DevExpress.ExpressApp

Namespace WinWebSolution.Module.Web
	Public Class WebShowCustomFormWindowController
		Inherits ShowCustomFormWindowController
		Private customFormPath As String = "CustomUserControls/WebCustomForm.aspx"
		Protected Overrides Sub ShowCustomNonModalForm()
			Dim page As Page = CType(Frame.Template, Page)

			Dim script As String = "<script>" & ControlChars.CrLf & "                                var openedWindow = window.open(""" & customFormPath & """,""test"", ""toolbar=no,menubar=no,width=600,height=600,resizable=no, center=yes"");" & ControlChars.CrLf & "                            </script>"
			page.ClientScript.RegisterStartupScript(Me.GetType(), "clientScriptForNonModalWindow", script)
		End Sub
		Protected Overrides Sub ShowCustomModalForm()
			Dim page As Page = CType(Frame.Template, Page)

			Dim script As String = "<script>" & ControlChars.CrLf & "                                window.showModalDialog(""" & customFormPath & """,""#1"",""dialogHeight: 600px; dialogWidth: 600px;dialogTop: 190px;  dialogLeft: 220px; edge: Raised; center: Yes;help: No; resizable: No; status: No;"");" & ControlChars.CrLf & "                            </script>"
			page.ClientScript.RegisterStartupScript(Me.GetType(), "clientScriptForModalWindow", script)
		End Sub
	End Class
End Namespace

File: CS/CS/WinWebSolution.Module.Web/WebShowCustomFormWindowController.cs
using System.Web.UI;
using DevExpress.ExpressApp;

namespace WinWebSolution.Module.Web {
    public class WebShowCustomFormWindowController : ShowCustomFormWindowController {
        string customFormPath = "CustomUserControls/WebCustomForm.aspx";
        protected override void ShowCustomNonModalForm() {
            Page page = (Page)Frame.Template;

            string script = @"<script>
                                var openedWindow = window.open(""" + customFormPath + @""",""test"", ""toolbar=no,menubar=no,width=600,height=600,resizable=no, center=yes"");
                            </script>";
            page.ClientScript.RegisterStartupScript(GetType(), "clientScriptForNonModalWindow", script);
        }
        protected override void ShowCustomModalForm() {
            Page page = (Page)Frame.Template;

            string script = @"<script>
                                window.showModalDialog(""" + customFormPath + @""",""#1"",""dialogHeight: 600px; dialogWidth: 600px;dialogTop: 190px;  dialogLeft: 220px; edge: Raised; center: Yes;help: No; resizable: No; status: No;"");
                            </script>";
            page.ClientScript.RegisterStartupScript(GetType(), "clientScriptForModalWindow", script);
        }
    }
}